### ISSUE TITLE
Improve tool chain - containers?

### ISSUE THREAD

**Issue by jol64** [FreeRDP/FreeRDP#5886]
**Is your feature request related to a problem? Please describe.**
This week I tried to build FreeRDP on linux and then Windows, and failed with both. The instructions at https://github.com/FreeRDP/FreeRDP/wiki/Compilation are probably both too scarce and outdated. I´d be willing to contribute occasionally, at least with testing, and at present I cannot with features unless they are already in a nightly build on master.

**Describe the solution you'd like**
A reproducable build environment that is easy to setup and ideally also get rid off. 

**Describe alternatives you've considered**
- my preference is clearly Docker, though LXC/LXD are fine as well. Even a full blown VM is ok.
- an automated setup script that pulls all tools used, sets up any tweeks (batch wrappers on windows), links, and does kind of validation it fits together.

Source should be outside of whatever container is used in order not to enforce any specific repository or editor. Cross compilation of wfreerdp on linux would be fine to me. Probably easier to handle then the myriad of tools today.

Ideally the outcome would be a network of build containers that reflects up- and downstream dependencies. FreeRDP is consuming OpenSSl and OpenH264, and is consume by Guacamole and others. I don´t want to build everything locally, ideally only the parts I need.

**Comment by akallabeth** [comment]
@jol64 well, if you have time we´d appreciate some help there.
But specifically the windows build is hard to script in such a way. (depending on your need you have to have openh264 and openssl dynamically or statically linked)
Currently the only reliable source is https://ci.freerdp.com but that has dependencies already set up (windows, mac and android side) or uses package builder tools (linux)

**Comment by jol64** [comment]
I am definitely willing to help, but I tried and failed. I tried to raise this via the mailing list, but I am obviously stuck in moderation...

"I tried with both Ubuntu 18.04 LTS server only and Debian 10.3.0 desktop (Gnome).
In both scenarios I failed with flatpak after configuration. Both flatpak install flathub org.freedesktop.Platform//18.08
And flatpak install flathub org.freedesktop.Sdk//18.08 yielded error messages about signatures not found.
Moreover, when I tried to continue despite the error messages libgstreamer appears not to be available as 0.10 anymore."

Anyone willing to support me, preferably peer to peer?

As long as I cannot build reliably, I have to wait until functionality is in nightly build or in a pull request build, but so far all of the pull request builds I looked at did not have any executables.
